manifester/helpers.py

```diff
@@ -113,13 +130,50 @@ def fetch_paginated_data(manifester, endpoint):
             total_results = len(_endpoint_data["body"])
             logger.debug(f"Total {endpoint} available on this account: {total_results}")
     if endpoint == "allocations":
-        return [
-            a for a in _endpoint_data["body"] if a["name"].startswith(manifester.username_prefix)
-        ]
+        if hasattr(_endpoint_data, "force_export_failure"):
+            return [
+                a
+                for a in _endpoint_data.allocation_data["body"]
+                if a["name"].startswith(manifester.username_prefix)
+            ]
+        else:
+            return [
+                a
+                for a in _endpoint_data["body"]
+                if a["name"].startswith(manifester.username_prefix)
+            ]
     elif endpoint == "pools":
         return _endpoint_data
 
 
+def load_inventory_file(file):
+    """Load local inventory file.
+
+    :return: list of dictionaries
+    """
+    if file.suffix in (".yaml", ".yml"):
+        loader = yaml
+        loader_args = {"Loader": yaml.FullLoader}
+        with file.open() as f:
+            return loader.load(f, **loader_args) or []
+    else:
+        logger.warn(
+            f"Found invalid inventory file {file}. Inventory file must exist and "
+            "have a .yaml or .yml suffix."
+        )
+
+
+def update_inventory(inventory_data):
+    """Replace the existing inventory file with current subscription allocations."""
+    inventory_path = Path(settings.inventory_path)
+    if load_inventory_file(inventory_path):
+        inventory_path.unlink()
+    inventory_path.touch()
+    if inventory_data != []:
+        with inventory_path.open("w") as inventory_file:
```

Comment:
just a check on logic here: if someone sync's their inventory and there is nothing for them, they will be left without an inventory file (as opposed to an empty file). Is this your intention?

Reply:
No, my intention is to leave a blank inventory file in place after a sync returns no results (i.e. an empty list). In my testing of this code, though, that is what consistently happens. Have you tested this and gotten different results?

Reply:
disregard the above, for some reason, I was reading unlink again instead of touch.